test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -615,12 +620,22 @@ class and its respective composed classes, and we don't reproduce all the error
 
     Tests both the run() and run_batch() methods.
     """
+    # These are extremely annoying to debug as they don't raise a stacktrace if a different exception type was thrown
+    # or let you know that it was an entirely different exception, or if the exception happened in the bulk version of the run
```

Comment:
In the java version of this code, it prints out the stacktrace if the exception type isn't the same - we could make the same update here. I just haven't needed it yet. You'd just add the stringified exception as the second argument to `assert`.

> or let you know that it was an entirely different exception

I don't understand this part - `assert type(got) == type(expected)` should tell you the two exception types are different

Reply:
Here's an example


```
tests_for_integration/api_to_db_test.py:1608 (test_run_job_batch_fail_no_allowed_wsid)
('Workspace ids are not allowed in RunJobParams in Batch Mode',) != (InvalidParameterForBatch('Workspace ids are not allowed in RunJobParams in Batch Mode'),)

Expected :(InvalidParameterForBatch('Workspace ids are not allowed in RunJobParams in Batch Mode'),)
Actual   :('Workspace ids are not allowed in RunJobParams in Batch Mode',)
<Click to see difference>

ee2_port = '49193'

    def test_run_job_batch_fail_no_allowed_wsid(ee2_port):
        params = [
            {"method": _MOD},
            {"method": _MOD, "wsid": 1},
        ]
        # this error could probably use some cleanup
        err = InvalidParameterForBatch("Workspace ids are not allowed in RunJobParams in Batch Mode")
>       _run_job_batch_fail(ee2_port, TOKEN_NO_ADMIN, params, {"wsid": 1}, err)

api_to_db_test.py:1616: 
```
When I called the function wrong

```
    err = InvalidParameterForBatch("Workspace ids are not allowed in RunJobParams in Batch Mode")
    _run_job_batch_fail(ee2_port, TOKEN_NO_ADMIN, params, {"wsid": 1}, err)
```
instead of
```
    err = "Workspace ids are not allowed in RunJobParams in Batch Mode"
    _run_job_batch_fail(ee2_port, TOKEN_NO_ADMIN, params, {"wsid": 1}, err)
```

Reply:
That seems pretty straightforward - it's telling you it returned a string rather than an exception
```

Expected :(InvalidParameterForBatch('Workspace ids are not allowed in RunJobParams in Batch Mode'),)
Actual   :('Workspace ids are not allowed in RunJobParams in Batch Mode',)
```

Reply:
<img width="734" alt="image" src="https://user-images.githubusercontent.com/1258634/125664143-6681efc2-4a26-44e3-a0b4-37f348c64824.png">

<img width="734" alt="image" src="https://user-images.githubusercontent.com/1258634/125664662-1c4b14f1-e0ad-4a2f-9d02-9de36bf3ca1e.png">

I was able to reproduce what I was talking about here

Reply:
E.g. you don't know which part of the test is throwing the exception. The fix for that would be

> In the java version of this code, it prints out the stacktrace if the exception type isn't the same - we could make the same update here. I just haven't needed it yet. You'd just add the stringified exception as the second argument to assert.